### PR TITLE
Add release notes: DSP, ztest, emul

### PR DESCRIPTION
**Pull request by yperess**
Document changes to the ztest Kconfigs, emul APIs, and the introduction of the DSP subsystem.

### CODE REVIEW

File: doc/releases/release-notes-3.3.rst [hunk 1]
```diff
@@ -1444,6 +1457,9 @@ Libraries / Subsystems
   * Added the linear range API to map values in a linear range to a range index
     :zephyr_file:`include/zephyr/sys/linear_range.h`.
 
+* DSP subsystem is now supported along with CMSIS as the default backend.
```
Comment (stephanosio) [inline doc/releases/release-notes-3.3.rst:1460]: "DSP" needs to be a dedicated section. This also needs to be alphabetically sorted

```suggestion
* DSP

  * Introduced DSP (digital signal processing) subsystem with CMSIS-DSP as the default backend.
  * CMSIS-DSP now supports all architectures supported by Zephyr.
```

**Review by stephanosio — CHANGES_REQUESTED** [review]
Looks good overall. Some comments about the DSP subsys release notes.

File: doc/releases/release-notes-3.3.rst (hunk 1 above)
Comment (yperess) [reply doc/releases/release-notes-3.3.rst]: Done

File: doc/releases/release-notes-3.3.rst [hunk 2]
```diff
@@ -1088,6 +1101,12 @@ Libraries / Subsystems
   * ``CONFIG_HAS_EXTERNAL_CACHE`` has been renamed to
     :kconfig:option:`CONFIG_EXTERNAL_CACHE`
 
+* DSP
+
+  * Introduced DSP (digital signal processing) subsystem with CMSIS-DSP at the
```
Comment (stephanosio) [inline doc/releases/release-notes-3.3.rst:1106]:
```suggestion
  * Introduced DSP (digital signal processing) subsystem with CMSIS-DSP as the
```

File: doc/releases/release-notes-3.3.rst [hunk 3]
```diff
@@ -17,6 +17,10 @@ Security Vulnerability Related
 API Changes
 ***********
 
+* Emulator creation APIs have changed to better match
+  :c:macro:`DEVICE_DT_DEFINE`. It also includs a new backend API pointer to
```
Comment (kartben) [inline doc/releases/release-notes-3.3.rst:21]: includs --> includes

**Review by stephanosio — DISMISSED** [review]

File: doc/releases/release-notes-3.3.rst [hunk 4]
```diff
@@ -1088,6 +1101,12 @@ Libraries / Subsystems
   * ``CONFIG_HAS_EXTERNAL_CACHE`` has been renamed to
     :kconfig:option:`CONFIG_EXTERNAL_CACHE`
 
+* DSP
+
+  * Introduced DSP (digital signal processing) subsystem with CMSIS-DSP as the
+    default backend.
+  * CMSIS-DSP now supports all architectures supported by Zephyr.
```
Comment (nordicjm) [inline doc/releases/release-notes-3.3.rst:1108]: Note needed on renaming to zdsp?

Comment (stephanosio) [reply doc/releases/release-notes-3.3.rst:1108]: No, CMSIS-DSP has not been renamed to ZDSP. It is just the default backend for the ZDSP (DSP subsystem).

**Review by dkalowsk — DISMISSED** [review]

**Comment by stephanosio** [comment]
@yperess This needs a rebase.

**Review by stephanosio — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by dkalowsk — APPROVED** [review]

**Pull request merged**